src/java_bytecode/java_object_factory.cpp

```diff
@@ -863,6 +862,19 @@ void java_object_factoryt::gen_nondet_struct_init(
         true,    // allow_null always true for sub-objects
         depth,
         substruct_in_place);
+
+      if(name=="length" &&
+          (class_identifier=="java.lang.String" ||
+           class_identifier=="java.lang.StringBuffer" ||
+           class_identifier=="java.lang.StringBuilder"))
```

Comment:
Do we also need to care about character arrays and `CharSequence` here?

Reply:
I believe these are the only types that we are supporting at the moment, but @romainbrenguier might answer that question more thoroughly.

Reply:
`CharSequence` is also supported but since it is abstract I didn't know if it made sense to initialize an object of that class in `java_object_factory`, it probably does so we could add it here.

Reply:
If we're not sure whether we can have `CharSequence` here, we should probably add an assertion. If we do, we might also want to check that we don't have `AbstractStringBuilder` either.

Reply:
Added invariant after those ifs preventing instantiation of those abstract types.

Reply:
Can you check that this will not be violated with a Java program where a function takes a `CharSequence` as argument?